tests/trinity/core/p2p-proto/test_peer_block_header_request_and_response_api.py

```diff
@@ -94,11 +95,13 @@ def mk_header_chain(length):
 @pytest.mark.asyncio
 async def test_les_peer_get_headers_round_trip(les_peer_and_remote,
                                                params,
+                                               monkeypatch,
                                                headers):
     peer, remote = les_peer_and_remote
+    request_id = 9
+    monkeypatch.setattr(trinity.protocol.les.managers, 'gen_request_id', lambda: request_id)
```

Comment:
I just realized that we should be able to get rid of this monkeypatching

In the `send_headers` function we can read the `GetBlockHeaders` request from the `remote` peer and extract the `request_id` from the message itself.  Take a look in the peer `_run` loop for the message handler.  This is probably easiest done using the `with peer.collect_sub_proto_messages` API.

Reply:
Hm, I thought I could just `await remote.read_msg()` but I'm getting a decryption error. I don't want to chase that down right now.